kerascv/layers/anchor_generators/anchor_generator.py

```diff
@@ -1,7 +1,47 @@
+# Copyright 2020 The Keras CV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
 import tensorflow as tf
 
 
 class AnchorGenerator(tf.keras.layers.Layer):
```

Comment:
Was this base class designed over the experience of https://github.com/tensorflow/models/blob/master/research/object_detection/core/anchor_generator.py? Or is this a from scratch design?

Reply:
Good question!
This is mostly a scratch design for proof of concept at this moment. Francois and I will send out RFC for public comments and feedbacks later :-)

Reply:
Ok cause there are a few derived classes at https://github.com/tensorflow/models/tree/master/research/object_detection/anchor_generators

Reply:
Yep and we intend to make them layers. Though the arguments are slightly different than this directory